src/version/features/UserTour/styles.js

```diff
@@ -1,7 +1,5 @@
-import { DEFAULT_MAIN_COLOR, DEFAULT_DANGER_COLOR } from "../../../core/config/styles";
-
-const themeColor = (window && window.config) ? window.config.mainColor : DEFAULT_MAIN_COLOR;
-const dangerColor = (window && window.config) ? window.config.dangerColor : DEFAULT_DANGER_COLOR;
+const themeColor = (window && window.config) ? window.config.theme.mainColor : "#0D672F";
```

Comment:
again, you should not access the window.config.theme here. It should only be accessed when you create the mui theme.

Reply:
@fzaninotto 

This file includes styles for Joyride-component, which is used for User Tour functionality:
```
// UserTour/styles.js
export default {
    options: {
         ....
    },
    tooltipTitle: {
         ....
    },
}

//UserTour/fragments/RunTourButton.js
import Joyride from 'react-joyride';
import toursStyles from "../styles";

            <Joyride
                ....
                styles={toursStyles}
            />
```

I tried to use Material-UI function withStyles() here and set styles via something like:
```
// UserTour/styles.js
export default theme => ({
    options: {
         ....
    },
    tooltipTitle: {
         ....
    },

});
```
But this variant doesn't work. For this reason can't use theme settings here.

Reply:
USe a custom component syles with withStyles() and inject it to Joyride, see https://docs.react-joyride.com/custom-components

Reply:
Unfortunately, not all components can be override by this way (look Props Inheritance):
https://docs.react-joyride.com/step

We can't override buttons, where we want to change the color by the theme like 
```
// ButtonNext.js
const styles = theme => ({});

const ButtonNext = ({ classes, ...props }) => {
     // our button
};

export default withStyles(styles)(ButtonNext);

 // RunTourButton.js
const RunTourButton = ({ classes, ...rest }) => {
    return (
       <Joyride
         ....
         buttonNextComponent={<ButtonNext {...props} />}
         ....
       />
    );
};
```

We can do following:
```
const styles = {
     ...
     ...
};

const JoyRideWithStyles = ({ classes, ...rest }) => {
    return (
              <Joyride
                continuous
                disableOverlayClose={true}
                showSkipButton={true}
                showProgress={true}
                locale={locale}
                steps={tourSteps}
                run={isPassed}
                styles={{classes}}
                callback={callback}
            />
    );
}

export default withStyles(styles)(JoyRideWithStyles);
```

But it will not work. Because withStyles() returns object with class names. Joyride doesn't use class names of its elements (buttons, tooltips etc.):
![usertour](https://user-images.githubusercontent.com/39263134/53819042-d5109980-3f71-11e9-969a-eec5958a06eb.png)